### DISCUSSION TITLE
Sonic 4 demo - object / clone related issues / errors

### DISCUSSION THREAD

**Discussion by 101Corp** [leopard-js/leopard-mentors#13] [Cleanup Crew]
### Scratch Project URL

https://scratch.mit.edu/projects/1046873489

### Leopard Project URL

https://leopardjs.com/projects/Mp1e1hBUmI

### What kind of error(s) are you experiencing?

- [ ] Leopard is failing to generate JavaScript code
- [X] The JavaScript code generated by Leopard doesn't work correctly
- [ ] The JavaScript code generated by Leopard is confusing or needs improvement
- [ ] Other

### Description of issue(s)

When collecting objects (this is a Sonic game, so rings) a glitchy effect may appear and all the objects will delete themselves instead of just the one you collected. This is detrimental because the objects are needed to progress. I hope this can be fixed.

**Comment by PullJosh** [comment]
Thanks for sharing your issue! 😄 Could you please hit "share" on your Leopard project so that we can take a look?

**Comment by PullJosh** [comment]
I just converted the project myself, and although I haven't looked into it deeply, I suspect that the problem is that Leopard does not support the "stop other scripts in sprite" block. You're using it in the `Objects` sprite and it doesn't work in Leopard:

<img width="580" alt="image" src="https://github.com/user-attachments/assets/510caf7f-eb0b-4698-bb80-d39102e93d5e">

The green comment (line 120) shown above is a TODO for us, the creators of Leopard. It should be implemented, but isn't yet.

In the meantime, if you can modify your Scratch code so that it works without using that block, your project should work. 😄

**Reply by 101Corp** [reply]
> I just converted the project myself, and although I haven't looked into it deeply, I suspect that the problem is that Leopard does not support the "stop other scripts in sprite" block. You're using it in the `Objects` sprite and it doesn't work in Leopard:
> 
> <img alt="image" width="580" src="https://private-user-images.githubusercontent.com/5458180/348438845-510caf7f-eb0b-4698-bb80-d39102e93d5e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4NDA4OTAsIm5iZiI6MTcyMDg0MDU5MCwicGF0aCI6Ii81NDU4MTgwLzM0ODQzODg0NS01MTBjYWY3Zi1lYjBiLTQ2OTgtYmI4MC1kMzkxMDJlOTNkNWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTNUMDMxNjMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjBmMThiZjQzOWQyOTQ0Y2YyYTcyODlkM2ZlYzE0NzRlZWIxZjUxMDU2MjYxNzA1MWEzZTcyN2IwM2EwZGUxYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.rSYd0UbiNu7YkkTUpZOMBpzHJL9A-cDUg2YbAF4PDUI">
> The green comment (line 120) shown above is a TODO for us, the creators of Leopard. It should be implemented, but isn't yet.
> 
> In the meantime, if you can modify your Scratch code so that it works without using that block, your project should work. 😄

I just changed it so the animations work without using that block. The animations are now functional, but the big issue of all the objects deleting themselves is still present.
